messaging/src/main/java/org/axonframework/queryhandling/QueryUpdateEmitter.java

```diff
@@ -189,4 +189,10 @@
     default Set<SubscriptionQueryMessage<?, ?, ?>> activeSubscriptions() {
         return Collections.emptySet();
     }
+
+    /**
+     * Closes all local running subscription queries.
+     *
+     */
+    void closeActiveSubscriptions();
```

Comment:
I'd rename the method to `close()` or `shutdown()`, since the `QueryUpdateEmitter` already defines it's about subscription queries.

Furthermore, it might be smart to already define the API to be async by returning a `CompletableFuture<Void>`.
The current implementation doesn't do anything funky which takes long, but future versions of the emitter might just block the shutdown process otherwise.